WordPress/gutenberg
# Don't add root padding to children of flex and grid layout blocks.

## What?


Fixes #52721 and #50861.

Modifies the root padding rules to not add root padding to children of full width flex and grid layout blocks. 

This change means that there will now be no space between children of full width flex and grid blocks and edges of the window. It might mean adding some extra margin to the parent block if this is not the desired outcome, but that is probably better than having padding on every single child block, which throws off the block spacing.

## Testing Instructions





1. Add a Row block to a post and make it full width;
2. Verify that its children don't have root padding in the editor or in the front end;
3. Activate the Grid experiment in Gutenberg > Experiments;
4. Add a Grid block to a post and make it full width;
5. Verify that its children don't have root padding in the editor or in the front end;
6. Convert the Row to a Group block and uncheck its "Inner blocks use content width" setting;
7. Verify that child blocks now have root padding in both editor and front end.

### Testing Instructions for Keyboard


## Screenshots or screencast
